### PR TITLE
Ch4 warning fixes

### PR DESCRIPTION
**Pull request by minsii**
This PR fixes several warnings when enable strict build. Some of the warnings were introduced by the uninlining commit.

### CODE REVIEW

**Comment by minsii** [comment]
test:jenkins/ch4/most

**Comment by minsii** [comment]
@akhillanger Could you please review the commit `mpi/coll: fix -Wunused-but-set-variable warning` for me ?
@yfguo Could you please review the other commits for me ?

**Comment by raffenet** [comment]
coll changes look fine as long as MPICH compiles. Akhil has moved on from the project, so no need to involve him.

**Comment by minsii** [comment]
@raffenet OK.

**Comment by minsii** [comment]
test:jenkins/ch4/most

File: src/mpid/ch4/src/ch4_init.c [hunk 1]
```diff
@@ -147,6 +147,12 @@ static int choose_netmod(void)
 #error "Thread Granularity:  Invalid"
 #endif
 
+static const char *MPIDI_CH4_mt_model_names[MPIDI_CH4_NUM_MT_MODELS] = {
```
Comment (yfguo) [inline src/mpid/ch4/src/ch4_init.c]: Since this array is used only in this file, can you remove its name prefix. `mt_model_names` should be sufficient.

Comment (minsii) [reply src/mpid/ch4/src/ch4_init.c]: Fixed

**Comment by minsii** [comment]
test:jenkins/ch4/most

**Comment by minsii** [comment]
Jenkins tests have passed. 
https://jenkins.mpich.org/view/review-ch4/job/mpich-review-ch4-ofi/2197/
https://jenkins.mpich.org/view/review-ch4/job/mpich-review-ch4-ucx/1933/

**Comment by minsii** [comment]
@yfguo It is ready for your review again. Could you please confirm ?

**Comment by yfguo** [comment]
@minsii The PR looks good. Please go ahead add my signoff and merge.

**Pull request merged**